src/common/PriorityCache.h

```diff
@@ -98,6 +98,9 @@ namespace PriorityCache {
     // Set the ratio of available memory this cache should target.
     virtual void set_cache_ratio(double ratio) = 0;
 
+    // If cache wants to get a little extra memory
+    virtual bool wants_extra() const = 0;
```

Comment:
Might just be easier to have a constructor param on ```PriorityCache::Manager``` to enable/disable this functionality. That would avoid the need to iterate through hundreds or thousands of journal player caches on rebalance.

Reply:
@dillaman In general case I can easily imagine a mix of caches which want a different behavior here. And I think that instead of this method we might event want a method that would calculate how much this particular cache wants. And I don't think iterating here will have any measurable effect to total time of rebalance.

On the other hand taking that this interface may still be change in the future I am ok with your suggestion too.